client/core/core.go

```diff
@@ -3367,39 +3441,6 @@ out:
 			// Queue the handling of this message.
 			nextJob <- &msgJob{handler, msg}
 
-		case <-ticker.C:
-			sinceLast := time.Since(lastTick)
-			lastTick = time.Now()
-			if sinceLast >= 2*dc.tickInterval {
-				// The app likely just woke up from being suspended. Skip this
-				// tick to let DEX connections reconnect and resync matches.
-				log.Warnf("Long delay since previous trade check (just resumed?): %v. "+
-					"Skipping this check to allow reconnect.", sinceLast)
-				continue
-			}
-			updatedAssets := make(assetMap)
-			dc.tradeMtx.Lock()
-			for oid, trade := range dc.trades {
-				if !trade.isActive() {
-					trade.mtx.Lock()
-					log.Infof("Retiring inactive order %v in status %v", oid, trade.metaData.Status)
-					trade.returnCoins()
-					trade.mtx.Unlock()
-					delete(dc.trades, oid)
-					updatedAssets.count(trade.wallets.fromAsset.ID)
-					continue
-				}
-				newUpdates, err := trade.tick()
-				if err != nil {
-					log.Error(err)
-				}
-				updatedAssets.merge(newUpdates)
-			}
-			dc.tradeMtx.Unlock()
-			if len(updatedAssets) > 0 {
-				dc.refreshMarkets()
-				c.updateBalances(updatedAssets)
-			}
```

Comment:
This does not need to be in the same loop as the incoming messages.  The tick operations need not delay message handling.  This is less of an issue with `prepareTrackedTrade` now holding the `dc.tradeMtx` more briefly, but it's still unnecessary in this loop.